back-end/helpers/test-helpers/testIntentGenerator.js

```diff
@@ -0,0 +1,97 @@
+export default class testIntentGenerator{
```

Comment:
This class does not properly follow the idea of a factory. Essentially, we want to create very generic methods that intake parameters and generate objects. A very simple factory might look like:
```
class MyFactory {
   generateWhatIWantToGenerate(param1, param2, param3) {
     objectThatIWantToGenerate = {
      key1: param1
      key2: param2
      key3: param3
      function: function (in this case jest.fn() of some kind)
     }
   }
}
```

Essentially, your methods here are too specific